backend/src/chats/chat-user-permissions/entities/chat-user-permission.entity.ts

```diff
@@ -0,0 +1,46 @@
+import {
+	Column,
+	Entity,
+	Index,
+	JoinColumn,
+	ManyToOne,
+	PrimaryGeneratedColumn,
+} from 'typeorm';
+import { User } from '../../../users/user/entities/user.entity';
+import { Chat } from '../../chat/entities/chat.entity';
+import { Permission } from '../../permissions/entities/permission.entity';
+
+@Entity()
+@Index(['chat_id', 'user_id', 'permission_id'], { unique: true })
+export class ChatUserPermission {
+	@PrimaryGeneratedColumn()
```

Comment:
this is from the documentation, here:
[many-to-many-relations-with-custom-properties](many-to-many-relations-with-custom-properties)

So it has a primary generated column because they do it in the example.

from the postgres docs:
https://www.postgresql.org/docs/current/ddl-constraints.html#DDL-CONSTRAINTS-PRIMARY-KEYS

> A primary key constraint indicates that a column, or group of columns, _can be used as a unique identifier for rows in the table_. This requires that the values be both unique and not null. 

and

> A table can have at most one primary key. [...] Relational database theory dictates that every table must have a primary key. This rule is not enforced by PostgreSQL, but it is usually best to follow it.

So 2 primary keys on both user_id and chat_id would not really do the trick. Since separately they aren't unique identifiers to the row. (together they might be, but not in the current implementation with multiple rows for permissions). So I think (now, I did not know before) that adding it to user_id and chat_id would not work as intended.

Reply:
prima